src/ImageSharp/Formats/Gif/GifDecoderCore.cs

```diff
@@ -506,14 +508,20 @@ private void RestoreToBackground(ImageBase<TPixel> frame)
         }
 
         /// <summary>
-        /// Sets the frame delay in the metadata.
+        /// Sets the frames metadata.
         /// </summary>
         /// <param name="metaData">The meta data.</param>
-        private void SetFrameDelay(IMetaData metaData)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void SetFrameMetaData(IMetaData metaData)
         {
             if (this.graphicsControlExtension != null && this.graphicsControlExtension.DelayTime > 0)
```

Comment:
You probably want to remove the second check from this if? You moved it inside already.